### PR TITLE
Enhancement/1492 dao identifier query param

### LINKED ISSUE #1492: Moving DAO identifier to query params
Related to https://github.com/decentdao/fractal-interface/issues/1483

But this one comes first.

Step 1 in this process is to move the location of the DAO's address from the path parameter, to a query parameter.

The accompanying PR for this issue should:
- Remove the `/daos` path prefix from all DAO-specific routes.
- Move the DAO address identifier from within the URL's `path`, to in a new `queryParam` (called "`dao`").
- Introduce a new path for the DAO's Dashboard (which used to be just the address of the DAO), called "`/home`".
- Maintain the URL path matching for existing links that are out there on the internet.

For example, a DAO's URL which is currently:
1. home: https://app.fractalframework.xyz/daos/0x36bD3044ab68f600f6d3e081056F34f2a58432c4
2. treasury: https://app.fractalframework.xyz/daos/0x36bD3044ab68f600f6d3e081056F34f2a58432c4/treasury
3. proposals: https://app.fractalframework.xyz/daos/0x36bD3044ab68f600f6d3e081056F34f2a58432c4/proposals

Should become:
1. home: https://app.fractalframework.xyz/home?dao=0x36bD3044ab68f600f6d3e081056F34f2a58432c4
2. treasury: https://app.fractalframework.xyz/treasury?dao=0x36bD3044ab68f600f6d3e081056F34f2a58432c4
3. proposals: https://app.fractalframework.xyz/proposals?dao=0x36bD3044ab68f600f6d3e081056F34f2a58432c4

### PR DESCRIPTION
**Pull request by adamgall**
Closes #1492

I decided to split the work from #1483 into two PRs, this being the first. To that effect, I created a new issue (#1492), which this PR closes.

This PR:
- Removes the `/daos` path prefix from all DAO-specific routes.
- Moves the DAO address identifier from within the URL's `path`, to in a new `queryParam` (called "`dao`").
- Introduces a new path for the DAO's Dashboard (which used to be just the address of the DAO), called "`/home`".
- Maintains the URL path matching for existing links that are out there on the internet.

For example, a DAO's URL used to be
1. home: https://app.fractalframework.xyz/daos/0x36bD3044ab68f600f6d3e081056F34f2a58432c4
2. treasury: https://app.fractalframework.xyz/daos/0x36bD3044ab68f600f6d3e081056F34f2a58432c4/treasury
3. proposals: https://app.fractalframework.xyz/daos/0x36bD3044ab68f600f6d3e081056F34f2a58432c4/proposals

But as per this PR, it will be
1. home: https://app.fractalframework.xyz/home?dao=0x36bD3044ab68f600f6d3e081056F34f2a58432c4
2. treasury: https://app.fractalframework.xyz/treasury?dao=0x36bD3044ab68f600f6d3e081056F34f2a58432c4
3. proposals: https://app.fractalframework.xyz/proposals?dao=0x36bD3044ab68f600f6d3e081056F34f2a58432c4

Additionally, we make sure to preserve existing URLs that are out there on the internet. To do that, we still match the `/daos/:address/*` path, and automatically redirect to `/home?dao=:address`.

A side effect that reviewers of this PR should be aware of, is that if someone had bookmarked or previously linked to _any_ page within a DAO (like the treasury or proposals page), the redirect will take a user to the _home_ page in the new URL structure.

### CODE REVIEW

File: src/constants/routes.ts [hunk 1]
```diff
@@ -7,7 +7,6 @@ type RouteInfo = { relative: (...args: any) => string; path: string };
 type RouteIndex = { [key: string]: RouteInfo };
 export interface DAORoutes extends RouteIndex {
   dao: RouteInfo;
-  daos: RouteInfo;
```
Comment (adamgall) [inline src/constants/routes.ts:10]: Noticed that this wasn't being used anywhere in the codebase, so removed it.

File: src/constants/routes.ts [hunk 2]
```diff
@@ -22,58 +21,54 @@ export interface DAORoutes extends RouteIndex {
 }
 
 export const DAO_ROUTES: DAORoutes = {
-  daos: {
-    relative: () => '/daos',
-    path: 'daos/:address/*',
-  },
```
Comment (adamgall) [inline src/constants/routes.ts:28]: Noticed that this wasn't being used anywhere in the codebase, so removed it.

File: src/constants/routes.ts [hunk 3]
```diff
@@ -22,58 +21,54 @@ export interface DAORoutes extends RouteIndex {
...
   },
   proposals: {
-    relative: (daoAddress: string) => `/daos/${daoAddress}/proposals`,
+    relative: (daoAddress: string) => `/proposals?dao=${daoAddress}`,
     path: 'proposals',
   },
   proposal: {
     relative: (daoAddress: string, proposalId: string) =>
-      `/daos/${daoAddress}/proposals/${proposalId}`,
+      `/proposals/${proposalId}?dao=${daoAddress}`,
```
Comment (adamgall) [inline src/constants/routes.ts:50]: Proposal IDs are still living within the path structure, but that's gonna have to be ok for now.

File: src/constants/routes.ts [hunk 4]
```diff
@@ -22,58 +21,54 @@ export interface DAORoutes extends RouteIndex {
...
   },
   proposalTemplates: {
-    relative: (daoAddress: string) => `/daos/${daoAddress}/proposal-templates`,
+    relative: (daoAddress: string) => `/proposal-templates?dao=${daoAddress}`,
     path: 'proposal-templates',
   },
   proposalTemplate: {
     relative: (daoAddress: string, proposalTemplateKey: string) =>
-      `/daos/${daoAddress}/proposal-templates/${proposalTemplateKey}`,
+      `/proposal-templates/${proposalTemplateKey}?dao=${daoAddress}`,
```
Comment (adamgall) [inline src/constants/routes.ts:67]: Same with these "proposal template keys"

File: src/hooks/DAO/useDAOController.ts [hunk 5]
```diff
@@ -11,7 +11,9 @@ import { useFractalTreasury } from './loaders/useFractalTreasury';
 import { useGovernanceContracts } from './loaders/useGovernanceContracts';
 
 export default function useDAOController() {
-  const { daoAddress } = useParams();
```
Comment (adamgall) [inline src/hooks/DAO/useDAOController.ts:14]: The type of _this_ `daoAddress` is `string | undefined`.

File: src/hooks/DAO/useDAOController.ts [hunk 6]
```diff
@@ -11,7 +11,9 @@ import { useFractalTreasury } from './loaders/useFractalTreasury';
...
+  const [searchParams] = useSearchParams();
+  const daoAddress = searchParams.get('dao');
```
Comment (adamgall) [inline src/hooks/DAO/useDAOController.ts:15]: The type of _this_ `daoAddress` is `string | null`.

File: src/hooks/DAO/useDAOController.ts [hunk 7]
```diff
@@ -24,7 +26,7 @@ export default function useDAOController() {
     }
   }, [action, daoAddress]);
 
-  const { nodeLoading, errorLoading } = useFractalNode({ daoAddress });
+  const { nodeLoading, errorLoading } = useFractalNode({ daoAddress: daoAddress || undefined });
```
Comment (adamgall) [inline src/hooks/DAO/useDAOController.ts:29]: So, to conform with `useFractalNode` expecting `string | undefined`, just did this to coerce the `null` to `undefined`.

File: src/hooks/DAO/useDAOController.ts
Comment (adamgall) [inline src/hooks/DAO/useDAOController.ts:1]: This file is the entry point to it all -- it's where we _used_ to grab the DAO address from the path, but _now_ grab the DAO address from a query param.

File: src/pages/DAOController.tsx
Comment (adamgall) [inline src/pages/DAOController.tsx:1]: The changes in this file were necessary to remove the `/daos` path prefix. This `DAOController`, which used to not "wrap" any components, now will end up wrapping the `<Layout>` component in the top level of the router.

File: src/router.tsx [hunk 8]
```diff
@@ -34,69 +36,69 @@ export const router = createBrowserRouter([
         element: <HomePage />,
       },
       {
-        path: '/create',
```
Comment (adamgall) [inline src/router.tsx:37]: Don't need that slash

File: src/utils/azorius.ts [hunk 9]
```diff
@@ -103,7 +103,7 @@ export const getProposalVotes = async (
       ...rest,
       voter,
       choice: VOTE_CHOICES[voteType],
-    } as ProposalVote; // This bypasses the type check, but it's fine
+    };
```
Comment (adamgall) [inline src/utils/azorius.ts:106]: Ugh this again!

Comment (Da-Colon) [reply src/utils/azorius.ts:106]: Such a annoying p3 issue. Like worth investigating? idk plenty of other things to do...

**Review by Da-Colon — COMMENTED** [review]
Code Review - Approval pending some live testing.

**Comment by adamgall** [comment]
I'm seeing an issue with this code.

If you attempt to load an invalid DAO, the "Invalid DAO" page is rendering without the context of the `<Layout>`, which is a bug.

File: src/pages/DAOController.tsx
Comment (adamgall) [reply src/pages/DAOController.tsx:1]: this comment is irrelevant now

File: src/utils/azorius.ts (hunk 9 above)
Comment (adamgall) [reply src/utils/azorius.ts:106]: idk, I'm not super concerned about it at the moment.

**Review by tomstuart123 — APPROVED** [review]
@adamgall - approval given

I did lots of testing of this PR with Decent, shutter, and myosin by pasting old URLs into this deploy link
Good news is they were all handled successfully 🙂

I can also confirm your point above that linking to an old link to the treasury page, will now link back to the core homepage (note - this is absolutely fine)

**Review by Da-Colon — APPROVED** [review]
Shit it

**Pull request merged**